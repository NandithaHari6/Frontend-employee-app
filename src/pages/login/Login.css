
.main-login-container{
  display: flex;

  height: 100vh;
}
.image-container{
    width:50%;
          display: flex;
  align-items: center;
  justify-content: center;
   height: 100vh;
   background-color: #7aceec;
}

.left-container{
    height:100%;
    width:50%;
}
.right-container{
    width:50%;
    height:100%;
    display:flex;
    justify-content: center;
    align-items: center;
}
.blue-div{
    margin:20px;
    background-color:#7aceec;
    height: 100%;;
    display:flex;
     justify-content: center;
    align-items: center;
}

.white-div{
    background-color: white;
    height: 77%;
    width:77%;
    border-radius: 50%;
        display:flex;
     justify-content: center;
    align-items: center;
};
.right-container{
    display: flex;
    justify-content: center;
    align-items: center;
}
.login-container{
    display:flex;
    flex-direction: column;
    gap:15px;
    width:70%;
}
.login-img{
height: 50%;
width: 50%;
object-fit: cover;
}
.logo-img{
  height: 40px;
    width: 180px;
}
.show-password{
    display:flex;
}