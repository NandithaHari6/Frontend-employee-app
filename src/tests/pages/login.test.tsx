import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import userEvent from "@testing-library/user-event";
import Login from "../../pages/login/login";
import baseApi from "../../api-services/api";

// Mock react-router-dom's useNavigate
const mockNavigate = vi.fn();
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});
// Mock the store
const mockStore = configureStore({
  reducer: {
    employee: (state = { employees: [] }) => state,
    department: (state = { departments: [] }) => state,
    [baseApi.reducerPath]: baseApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(baseApi.middleware)
});
// Mock the login mutation
const mockLoginMutation = vi.fn();
vi.mock("../../api-services/auth/login.api", () => ({
  useLoginMutation: () => [
    () => ({
      unwrap: () => mockLoginMutation(),
    }),
    { isLoading: false },
  ],
}));
describe("Login Page", () => {
  const renderLogin = () => {
    return render(
      <Provider store={mockStore}>
        <Login />
      </Provider>
    );
  };
  beforeEach(() => {
    vi.clearAllMocks();
    vi.spyOn(window.localStorage.__proto__, "getItem").mockImplementation(
      (key) => {
        if (key === "token") return "valid-token"; // simulate token present
        return null;
      }
    );
  });
  it("should match snapshot", () => {
    const { container } = renderLogin();
    expect(container).toMatchSnapshot();
  });

  it("should render login form with all required elements", () => {
    renderLogin();

    // Check if username input is present
    const usernameInput = screen.getByLabelText("Email");
    expect(usernameInput).toBeInTheDocument();

    // Check if password input is present
    const passwordInput = screen.getByLabelText("Password");
    expect(passwordInput).toBeInTheDocument();

    // Check if submit button is present
    const submitButton = screen.getByRole("button", { name: /login/i });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute("type", "button");
  });
  it("should show error when username is too long", async () => {
    renderLogin();
    const usernameInput = screen.getByLabelText("Email");
    // Type a username longer than 30 characters
    await userEvent.type(usernameInput, "a".repeat(31));
    // Check if error message is displayed
    expect(
      screen.getByText("Username error: greater than 10 characters")
    ).toBeInTheDocument();
  });


  
  it("should handle successful login", async () => {
    mockLoginMutation.mockResolvedValueOnce({ token: "fake-token" });

    renderLogin();

    // Fill in the form
    const usernameInput = screen.getByLabelText("Email");
    const passwordInput = screen.getByLabelText("Password");
    const submitButton = screen.getByRole("button", { name: /login/i });

    await userEvent.type(usernameInput, "test@example.com");
    await userEvent.type(passwordInput, "password123");

    // Submit the form
    await userEvent.click(submitButton);

    // Wait for the login mutation to resolve and navigation to occur
    await waitFor(() => {
      expect(mockLoginMutation).toHaveBeenCalled();
      expect(mockNavigate).toHaveBeenCalledWith("/employee");
    });
  });
});
