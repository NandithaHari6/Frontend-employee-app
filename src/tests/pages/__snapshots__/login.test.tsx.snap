// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match snapshot 1`] = `
<div>
  <div
    class="main-login-container"
  >
    <div
      class="left-container"
    >
      <div
        class="blue-div"
      >
        <div
          class="white-div"
        >
          <img
            alt="image"
            class="login-img"
            src="/assets/kv-login.jpeg"
          />
        </div>
      </div>
    </div>
    <div
      class="right-container"
    >
      <div
        class="login-container"
      >
        <img
          alt="logo"
          class="logo-img"
          src="/assets/kv-logo.png"
        />
        <div
          class="form-group"
        >
          <label
            class="for"
            for="Email"
          >
            Email
          </label>
          <div
            class="input-field"
          >
            <input
              aria-label="Email"
              class="input-box undefined"
              id="Email"
              placeholder="Enter your email"
              type="email"
              value=""
            />
            <button
              disabled=""
              type="button"
            >
              Clear
            </button>
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class="for"
            for="Password"
          >
            Password
          </label>
          <div
            class="input-field"
          >
            <input
              aria-label="Password"
              class="input-box undefined"
              id="Password"
              placeholder="Enter your password"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="show-password"
        >
          <input
            checked=""
            type="checkbox"
          />
          <p>
            Show Password
          </p>
        </div>
        <button
          class="common-btn + undefined"
          type="button"
        >
           
          Login
        </button>
      </div>
    </div>
  </div>
</div>
`;
